### PR TITLE
Migrate metricbeat docker module to ECS

### LINKED ISSUE #10900: Metricbeat docker module is not ECS ready
`add_docker_metadata` has been migrated to ECS (#9412) and docker autodiscover is going to be migrated too (#10757). For this, all `docker.container.*` fields except `docker.container.labels` are aliases now to `container.*` fields. We should migrate also the metricbeat docker module so it doesn't try to write to the old fields.

### PR DESCRIPTION
**Pull request by jsoriano**
Migrate docker metricsets with conflicting fields detected by #10921 to ECS.

Fix #10900

### CODE REVIEW

**Comment by kaiyan-sheng** [comment]
This PR also migrates docker module to use ReporterV2 interface. Please see #10774 for more info.

File: metricbeat/module/docker/container/data.go [hunk 1]
```diff
@@ -23,39 +23,49 @@ import (
...
+			},
 		},
-		"status": cont.Status,
 	}
 
 	labels := docker.DeDotLabels(cont.Labels, dedot)
 
 	if len(labels) > 0 {
-		event["labels"] = labels
+		event.Put("docker.container.labels", labels)
```
Comment (kaiyan-sheng) [inline metricbeat/module/docker/container/data.go]: Should this be `container.labels` to match ecs?

Comment (jsoriano) [reply metricbeat/module/docker/container/data.go]: We are not migrating `docker.container.labels` as we don't have a good upgrade path at the moment for these kind of fields. See https://github.com/elastic/beats/pull/9412#issuecomment-446887660

Comment (kaiyan-sheng) [reply metricbeat/module/docker/container/data.go]: ahh yes! You told me this last week and I forgot... sorry!

**Comment by ruflin** [comment]
@jsoriano Thanks for taking this on. Not sure why me missed this one in https://github.com/elastic/beats/issues/8655 :-(

**Pull request merged**